services/galaxy-toolshed/galaxy-toolshed-activity.service.js

```diff
@@ -0,0 +1,70 @@
+import Joi from 'joi'
+import { formatDate } from '../text-formatters.js'
+import { age as ageColor } from '../color-formatters.js'
+import BaseGalaxyToolshedService from './galaxy-toolshed-base.js'
+
+const repositoryRevisionInstallInfoSchema = Joi.array().items(
+  Joi.object({
+    create_time: Joi.string().required(),
+  }).required(),
+  Joi.object({}),
+  Joi.object({})
+)
+
+export default class GalaxyToolshedActivity extends BaseGalaxyToolshedService {
+  static category = 'activity'
+  static route = {
+    base: 'galaxy-toolshed/create-date',
+    pattern: ':repositoryName/:owner',
+  }
+
+  static examples = [
+    {
+      title: 'Galaxy Toolshed - Create Time',
+      namedParams: {
+        repositoryName: 'sra_tools',
+        owner: 'iuc',
+      },
+      staticPreview: this.render({
+        createDate: new Date(0).setUTCSeconds(1538288239),
+      }),
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'create date',
+  }
+
+  static render({ createDate }) {
+    return { message: formatDate(createDate), color: ageColor(createDate) }
+  }
+
+  async fetch({ repositoryName, owner }) {
+    const changesetRevisions =
+      await this.fetchOrderedInstallableRevisionsSchema({
+        repositoryName,
+        owner,
+      })
+    return this.fetchRepositoryRevisionInstallInfoSchema({
+      schema: repositoryRevisionInstallInfoSchema,
+      repositoryName,
+      owner,
+      changesetRevision: changesetRevisions.shift(),
+    })
+  }
```

Comment:
Having to make two sequential API calls like this is generally frowned upon. The reason is that we have a very limited window in which to complete the entire badge request/response flow in order to get the badge back to the user in time before the renderers in common places like GitHub timeout and refuse to show the badge.

I gather you're taking this sequential approach to first get the latest version and then fetch the stats for that specific version, but is there a different endpoint that would return the stats for all versions? It's a lot easier for us to do a tad more processing (and we've a ton of utility functions for this) as opposed to dealing with multiple networking round trips just to fetch data from the provider

Reply:
Fair enough. Once you get a chance to rebase we should be able to update the review app and can start getting a ballpark feel for the performance